### ISSUE TITLE
Project Health

### ISSUE THREAD

**Issue by stevekrouse** [stevekrouse/WoofJS#555]
* tests
  * research testing system
  * ensure .es6 == .js
  * ensure .es6 is style is good (prettier), pre-commit test?
  * web based automated testing?
  * tests for core #328
  * tests for /create
* benchmark speed load times & remove bloat
* build system research 
  * js style autoformat such as prettier
  * typescript research
  * research directory structure to split up /create into files
  * abstract /create & /team
  * benchmark & minification
* refactor
  * refactor all callbacks to promises, and async style
  * refactor ugly uses of codemirror / research monaco

**Comment by stevekrouse** [comment]
## Testing

[Everyone seems to think testing is a good idea.](https://testingjavascript.com/) Yet in my experience it takes a lot of time (50%), doesn't catch many bugs, and we still have to manually run things anyways. Of course this is only true on small projects with small user bases.

I think my policy will be that instead of testing everything, let's only put tests in place for things that break once. Yes this will require our users to be our bug testers, but I think that's ok as long as we can rollback quickly and fix bugs quickly.

In other words, let's not pre-optimize here. Let's see if we create too many problems first before over-engineering our testing suite.

If we do decide to add testing, cypress.io looks neat! These are the standard options: mochajs.org, jestjs.io

**Comment by stevekrouse** [comment]
## Simple Tests & Formatting

1. What would make everyone's life easier is if we were guaranteed that `.es6 == .js`. This could be a pre-commit hook, but that would require installation. It could be a test we run on pull request, but it could also be a *job* run on pull request to do it for you automatically and submit as a commit for you.

2. Another idea that will make everyone's life easier is automatic prettier formatting. Again this can happen on save, as a precommit hook, or as a bot that criticizes you or does it for you.

Seems like the [pre-commit hook is the standard way for prettier](https://prettier.io/docs/en/precommit.html)

I found this bot restyled.io but it didn't seem to do what I wanted... 

I could make my own bot like I did for the Zapier rawgit/githack thing with Probot https://probot.github.io/

Travisci or Circleci could be helpful here too...
